### ISSUE TITLE
Triggering assertion when using validation folder

### ISSUE THREAD

**Issue by Gennaro-Farina** [divamgupta/image-segmentation-keras#73]
Hi, 

I had to replace this line of code:
`assert ( seg in segmentations_d ),  (im + " is present in "+images_path +" but "+seg_bnme+" is not found in "+segs_path + " . Make sure annotation image are in .png"  )`

with this one: 
`assert (dict_contains_substr(segmentations_d, os.path.basename(seg) )),  (im + " is present in "+images_path +" but "+seg_bnme+" is not found in "+segs_path + " . Make sure annotation image are in .png"  )`

to get code running when using a validation set.

The replaced line of code uses this tiny improvable function:
```
def dict_contains_substr(dictionary, substr):
	for k in dictionary:
		if substr in k:
			return True
	for v in dictionary.items():
		if substr in v:
			return True
	return False
```

Was that a fault of mine?
